metasync.js

```diff
@@ -72,3 +72,28 @@ metasync.sequential = function(funcs, done) {
   if (len > 0) next();
   else done();
 };
+
+
+// Asynchrous filter
+// filter :: (a -> Boolean) -> [a] -> ([a] -> Void)
+metasync.filter = function(predicate, array, done) {
+  var result = [];
+  var index = 0;
+  
+  function doFilter() {
+    if (index === array.length) {
+      done(result);
+    } else {
+      if (predicate(array[index])) {
```

Comment:
Well, could you please provide some real-world like example of this function usage?